### DISCUSSION TITLE
Discord or community hub?

### DISCUSSION THREAD

**Discussion by seanwiththebeard** [sehugg/8bitworkshop#72] [Q&A] [answered]
I've been using your IDE to (re)learn C and code for C64 over the past six months or so, it's kind of addictive. Is there an official Discord server or message board for users to showcase their works-in-progress and talk about coding for 8-bit platforms?

**Comment by sehugg** [comment, marked as answer]
There's not yet -- I'd like to do something social with 8bitworkshop, but integrated with the IDE. Otherwise I don't think I could build a critical mass of users. People are spread pretty thin across social media and existing long-time forums.

Everything C64 might be a good place to go for now. They have a Discord server: https://discord.gg/52JAgfvdy2

Any other suggestions? I created this discussion board as an experiment, but the Issues tab seems more popular :)

**Reply by seanwiththebeard** [reply]
Right now, the thing that's going to hold me back is being able to mount a disk image to the C64 emulator. I'm writing an Ultima clone and getting close to where I need disk access to swap out maps and data (mapdemo1 on my GitHub if you want to see). Knowing that I'll eventually need disk access, I've been switching between 8bitworkshop and VS Code with the VICE debugger plugin, which has been helpful because I like having more files open in tabs and the IntelliSense in the editor, but the debugging and memory viewer is so much better on 8bitworkshop. Some kind of VS Code integration would be neat to see but I don't know what that would look like.

The biggest things I'd like to see on the roadmap are the disk image mounting and general settings for the C64 emulator to pick palette, NTSC/PAL, aspect ratio, etc.

**Comment by sehugg** [comment]
Disk support is tricky. The C64 emulator (chips) doesn't support it, MAME does but it's not built for C64 yet. Even so, every project seems to have special requirements for packers/loaders/etc and a custom toolchain seems inevitable. Though maybe the IDE could allow custom build steps written in JavaScript...?

Even if you use VSCode and a custom toolchain, you could still use 8bitworkshop for testing individual modules, for example:
```c
//#link "file_needed_for_test.c"
#ifdef __MAIN__
#include "file_needed_for_test.h"
void main() {
    test_my_module();
}
#endif
```
I've been working on a desktop version of 8bitworkshop that can access local files (hardcoded to ~/8bitworkshop) if you'd like to try it: https://8bitworkshop.com/release/desktop/
